src/details/ArborX_DetailsDistributedTreeUtils.hpp

```diff
@@ -107,24 +108,26 @@ void sortResults(ExecutionSpace const &space, View keys,
   if (n == 0)
     return;
 
-  // We only want to get the permutation here, but sortObjects also sorts the
-  // elements given to it. Hence, we need to create a copy.
-  // TODO try to avoid the copy
-  View keys_clone(
-      Kokkos::view_alloc(space, Kokkos::WithoutInitializing,
-                         "ArborX::DistributedTree::query::sortResults::keys"),
-      keys.size());
-  Kokkos::deep_copy(space, keys_clone, keys);
-  auto const permutation = ArborX::Details::sortObjects(space, keys_clone);
-
-  // Call applyPermutation for every entry in the parameter pack.
-  // We need to use the comma operator here since the function returns void.
-  // The variable we assign to is actually not needed. We just need something
-  // to store the initializer list (that contains only zeros).
-  auto dummy = {
-      (ArborX::Details::applyPermutation(space, permutation, other_views),
-       0)...};
-  std::ignore = dummy;
+  if constexpr (sizeof...(OtherViews) == 1 &&
+                std::tuple_element_t<0, std::tuple<OtherViews...>>::rank == 1)
+  {
+    // If there's only one 1D view to process, we can avoid computing the
+    // permutation.
+    KokkosExt::sortByKey(space, keys, other_views...);
```

Comment:
I would be interested in us implementing a variadic overload of sort by key which would allow us to just call that function unconditionally.
(Not asking to implement in this PR)